### ISSUE TITLE
eth_getbalance request never ends

### ISSUE THREAD

**Issue by laureveler** [WalletConnect/WalletConnectUnity#96]
I had this working on v2 prior to the recent updates.
Pseudo-code:

```
 [RpcMethod("eth_getBalance"), RpcRequestOptions(Clock.ONE_MINUTE, 99998)]
    public class EthGetBalance : List<string>
    {
        public EthGetBalance(string address, BigInteger? blockNumber = null) : base(new[] { address, blockNumber != null ? blockNumber.ToString() : "latest" }) { }
    }

async void GetBalanceRoutine 
{
      _request = new EthGetBalance(wallet.WalletAddress);
      BigInteger result = await WCSignClient.Instance.Request<EthGetBalance, BigInteger>(_wallet.Session.Topic, _request, _wallet.ChainId);     
      Debug.Log(result);
}
```

Now the await statement never ends.
Anyone had any luck with this?

**Comment by gigajuwels** [comment]
WalletConnect protocol does technically support this RPC **however**, this requires the dApp and wallet to agree on allowing this method(s) via required and optional namespaces. And according to @pedrouid 

> no wallet that (I know of) accepts [them] (`eth_getBalance`)

**Comment by adam-bloktopia** [comment]
Any updates on when a fix for this bug might be coming?

**Comment by skibitsky** [comment]
Most wallets still don't support `eth_getBalance`. We recommend sending this request directly to RPC node. Added explanation and code sample to the [WalletConnect Modal docs](https://docs.walletconnect.com/advanced/walletconnectmodal/usage?platform=unity#interaction-with-rpc).

tl;dr
* Install [Nethereum.Web3](https://www.nuget.org/packages/nethereum.web3)
* Use this code to call `eth_getBalance`
```csharp
private static async Task GetAccountBalance()
{
    var session = WalletConnect.Instance.ActiveSession;

    // Because one session have multiple namespaces, we need to select one.
    // In most cases, especially in games, dapp will use only one namespace.
    var @namespace = session.RequiredNamespaces.First();
    var address = session.CurrentAddress(@namespace.Key).Address;

    var config = ProjectConfiguration.Load();

    // Using WalletConnect Blockchain API: https://docs.walletconnect.com/cloud/blockchain-api
    var url = $"https://rpc.walletconnect.com/v1?chainId={@namespace.Value.Chains[0]}&projectId={config.Id}";

    var web3 = new Nethereum.Web3.Web3(url);
    var balance = await web3.Eth.GetBalance.SendRequestAsync(address);
    Debug.Log($"Balance of {address} in Wei: {balance.Value}");

    var etherAmount = Nethereum.Web3.Web3.Convert.FromWei(balance.Value);
    Debug.Log($"Balance of {address} in Ether: {etherAmount}");
}
```
